plugins/plugins/sentinel/config.go

```diff
@@ -0,0 +1,105 @@
+// Copyright The HTNN Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sentinel
+
+import (
+	sentinelApi "github.com/alibaba/sentinel-golang/api"
+	sentinelConf "github.com/alibaba/sentinel-golang/core/config"
+	"github.com/alibaba/sentinel-golang/logging"
+
+	"mosn.io/htnn/api/pkg/filtermanager/api"
+	"mosn.io/htnn/api/pkg/plugins"
+	"mosn.io/htnn/plugins/plugins/sentinel/rules"
+	"mosn.io/htnn/types/plugins/sentinel"
+)
+
+func init() {
+	plugins.RegisterPlugin(sentinel.Name, &plugin{})
+}
+
+type plugin struct {
+	sentinel.Plugin
+}
+
+func (p *plugin) Factory() api.FilterFactory {
+	return factory
+}
+
+func (p *plugin) Config() api.PluginConfig {
+	return &config{}
+}
+
+type config struct {
+	sentinel.CustomConfig
+
+	params      sentinelApi.EntryOption
+	attachments []*sentinel.Source
+	m           *res2RuleMap
+}
+
+type res2RuleMap struct {
+	f  map[string]*sentinel.FlowRule
+	hs map[string]*sentinel.HotSpotRule
+	cb map[string]*sentinel.CircuitBreakerRule
+}
+
+func (conf *config) Init(cb api.ConfigCallbackHandler) error {
+	sc := sentinelConf.NewDefaultConfig()
+	sc.Sentinel.Log.Logger = logging.NewConsoleLogger()
+	sc.Sentinel.Log.Dir = conf.GetLogDir()
```

Comment:
Should its default behavior be to output to the console or the default directory? I currently have the default log directory `/tmp/sentinel` set in Configuration Validate.

Reply:
It would be better not to write the disk by default.

Reply:
sentinel-golang logs come in two types: metric and record the log, see https://sentinelguard.io/zh-cn/docs/golang/logging.html. 

When we want the log output to the console, set `Log.Dir = ""` and `Log.Logger = logging.NewConsoleLogger()` to output the record log to the console as expected, but the metric log will still be output to the default directory (~/logs/csp). 

I think the inability of metric to output to the console is a defect of sentinel-golang (see https://github.com/alibaba/sentinel-golang/blob/master/core/log/metric/writer.go#L293), so my recommendation is to give the default directory `/tmp/sentinel` instead of output to the console when Log.Dir is empty

Reply:
What a pity! So we have to use a default directory. Could you add your investigation as a comment?

Reply:
Is it possible to disable metrics? Writing metrics without collector is meaningless.

Reply:
Disabling metric log is a good idea. After my exploration of the source code, disabling it is feasible (`Log.Metric.FlushIntervalSec = 0`)! That will fulfill your log expectations.

Reply:
Good. Maybe we can disable metrics & log to console by default? In this way, people don't need to persist files on the disk.